Batteries/Data/Stream.lean

```diff
@@ -0,0 +1,93 @@
+/-
+Copyright (c) 2024 François G. Dorais. All rights reserved.
+Released under Apache 2. license as described in the file LICENSE.
+Authors: François G. Dorais
+-/
+
+namespace Stream
+
+/-- Drop up to `n` values from the stream `s`. -/
+def drop [Stream σ α] (s : σ) : Nat → σ
+  | 0 => s
+  | n+1 =>
+    match next? s with
+    | none => s
+    | some (_, s) => drop s n
+
+/-- Read up to `n` values from the stream `s`. -/
+def take [Stream σ α] (s : σ) : Nat → List α × σ
```

Comment:
What is the rationale for `List` in place of `Array`

Reply:
Good question. I don't recall thinking about it much but it's a natural choice since `List` is a `Stream` but `Array` is not.

Reply:
I can't think of major applications that weigh one way or the other.